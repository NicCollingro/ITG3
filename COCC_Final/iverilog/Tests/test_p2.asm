.text
	ldi A 0x11
	ldi B 0x22
	ldi C 0x33
	ldi D 0x44
	ldi E 0x55
	ldi F 0x66
	ldi G 0x77

	rout A
	rout B
	rout C
	rout D
	rout E
	rout F
	rout G
	
	mout %gx
	mout %gy
	mout %gz

	ldi A 0
	ldi B 0
	ldi C 0
	ldi D 0
	ldi E 0
	ldi F 0
	ldi G 0

	rin A
	rin B
	rin C
	rout A
	rout B
	rout C

	min %gx
	min %gy
	min %gz
	mout %gx
	mout %gy
	mout %gz

	hlt

.data
	gx = 0xaa
	gy = 0xbb
	gz = 0xcc
