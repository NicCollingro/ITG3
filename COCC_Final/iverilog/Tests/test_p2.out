    4600 OUTPUT:  17 (0x11 0b00010001)
    5080 OUTPUT:  34 (0x22 0b00100010)
    5560 OUTPUT:  51 (0x33 0b00110011)
    6040 OUTPUT:  68 (0x44 0b01000100)
    6520 OUTPUT:  85 (0x55 0b01010101)
    7000 OUTPUT: 102 (0x66 0b01100110)
    7480 OUTPUT: 119 (0x77 0b01110111)
    8200 OUTPUT: 170 (0xaa 0b10101010)
    8920 OUTPUT: 187 (0xbb 0b10111011)
    9640 OUTPUT: 204 (0xcc 0b11001100)
   14320 INPUT:   22 (0x16 0b00010110)
   14800 INPUT:   23 (0x17 0b00010111)
   15280 INPUT:   24 (0x18 0b00011000)
   15760 OUTPUT:  22 (0x16 0b00010110)
   16240 OUTPUT:  23 (0x17 0b00010111)
   16720 OUTPUT:  24 (0x18 0b00011000)
   17440 INPUT:   27 (0x1b 0b00011011)
   18160 INPUT:   28 (0x1c 0b00011100)
   18880 INPUT:   29 (0x1d 0b00011101)
   19600 OUTPUT:  27 (0x1b 0b00011011)
   20320 OUTPUT:  28 (0x1c 0b00011100)
   21040 OUTPUT:  29 (0x1d 0b00011101)
REGISTERS: A: 16 B: 17 C: 18 D: 00 E: 00 F: 00 G: 00 H: xx
FLAGS:     CF: x ZF: x
